build.gradle.kts

```diff
@@ -56,6 +56,49 @@ tasks.jar {
     }
 }
 
+// GraalVM Native Image configuration
+tasks.register<Test>("testWithAgent") {
+    group = "verification"
+    description = "Runs all tests with GraalVM native-image agent to generate metadata."
+
+    val metadataOutputDir = layout.buildDirectory.dir("native-image-metadata").get().asFile
+    val resourcesTargetDir = layout.projectDirectory.dir("src/main/resources/META-INF/native-image/minestom").asFile
+    val filterFile = layout.buildDirectory.file("agent-filter.json").get().asFile
+
+    testClassesDirs = sourceSets["test"].output.classesDirs
+    classpath = sourceSets["test"].runtimeClasspath
+
+    // Configure the native-image agent
+    jvmArgs(
+            "-agentlib:native-image-agent=config-output-dir=${metadataOutputDir.absolutePath},caller-filter-file=${filterFile.absolutePath}"
+    )
+
+    doFirst {
+        delete(metadataOutputDir)
+        metadataOutputDir.mkdirs()
+
+        filterFile.parentFile.mkdirs()
+        filterFile.writeText("""
+{
+  "rules": [
+    {"excludeClasses": "org.junit.**"},
+    {"excludeClasses": "org.opentest4j.**"},
```

Comment:
seems like this either isnt working or needs to be regenerated since being added, since there are still targeted classes in the output.